service/history/transferQueueTaskExecutorBase.go

```diff
@@ -260,8 +260,19 @@ func (t *transferQueueTaskExecutorBase) deleteExecution(
 		return err
 	}
 
-	// Do not validate version here because DeleteExecutionTask transfer task is created only for
-	// explicit API call, and in this case execution needs to be deleted regardless of the version.
+	// Skip the task version check in two cases:
+	// 1. Request is from explicit API call
+	// 2. Local namespace
```

Comment:
```suggestion
	// If task version is EmptyVersion it means "don't check task version". This can happen when task was created from explicit user API call.
```

Reply:
updated